example/container_metric/example.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/rand"
+	"sync"
+	"time"
+
+	mexporter "github.com/GoogleCloudPlatform/opentelemetry-operations-go/exporter/metric"
+
+	"go.opentelemetry.io/otel/label"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/sdk/metric/controller/basic"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+type observedInt struct {
+	mu sync.RWMutex
+	f  int64
+}
+
+func (of *observedInt) set(v int64) {
+	of.mu.Lock()
+	defer of.mu.Unlock()
+	of.f = v
+}
+
+func (of *observedInt) get() int64 {
+	of.mu.RLock()
+	defer of.mu.RUnlock()
+	return of.f
+}
+
+func newObservedInt(v int64) *observedInt {
+	return &observedInt{
+		f: v,
+	}
+}
+
+func main() {
+	// Initialization. In order to pass the credentials to the exporter,
+	// prepare credential file following the instruction described in this doc.
+	// https://pkg.go.dev/golang.org/x/oauth2/google?tab=doc#FindDefaultCredentials
+	opts := []mexporter.Option{
+		mexporter.WithMetricDescriptorTypeFormatter(func(desc *metric.Descriptor) string {
+			return fmt.Sprintf("kubernetes.io/%s", desc.Name())
+		}),
+		mexporter.WithInterval(10 * time.Second),
+	}
+
+	// These are the semantic conventions for containers that map to the k8s_container resource
```

Comment:
Wouldn't it make more sense to use https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/detectors/gcp/gke.go here as an example closer to what we recommend to our users?

Reply:
we could potentially override all of the env vars for GKE, but the GCE ones won't be set unless the example is run inside GCE.  My hope was to have an example I could reference when I needed to test GKE metrics locally

Reply:
> but the GCE ones won't be set unless the example is run inside GCE.

GKE nodes are running on GCE VMs (well, for "GKE on GCP" specifically, not on-prem or AWS), so I thought they would be able to reach GCP metadata server and fetch GCE metadata from there?

The purpose of an example is usually for the end user to copy-paste and modify it to their needs, so my fear here is that instead of using auto-detection mechanism they will get the idea that they have to manually provide all the attributes every time.

Maybe consider adding a comment or adding some flag to the application to run in the "local" mode vs "production" mode on GCE/GKE?

Reply:
Yeah, I think it might be better just to keep this example for myself to reference, and keep existing examples for users.